### ISSUE TITLE
Inf and nan loss 

### ISSUE THREAD

**Issue by ayush1999** [pytorch/pytorch#6126]
Here is my network:
```
import torch.nn as nn
from torch.autograd import Variable

class RNN(nn.Module):
    def __init__(self, input_size, hidden_size, output_size):
        super().__init__()
        self.hidden_size = hidden_size
        self.i2h = nn.Linear(input_size , hidden_size)
        self.h2o = nn.Linear(hidden_size, output_size)
        self.h2h = nn.Linear(hidden_size, hidden_size)
        self.softmax = nn.LogSoftmax(dim = 1)

    def forward(self, input, hidden):
        h = self.softmax(self.h2h(hidden)+  self.i2h(input))
        o = self.softmax(self.h2o(h))
        return o, h

    def init_hidden(self):
        return Variable(torch.zeros(1, self.hidden_size))

```
Then, I created the network and the training loops as :
```

rnn = RNN(n_chars, 90, n_chars)
criterion = nn.KLDivLoss()
learning_rate = 0.05
optimizer = torch.optim.Adam(rnn.parameters(), lr = learning_rate)
hidden = rnn.init_hidden()
epochs = 5

for epoch in range(epochs):
    for i in range(len(X)):
        for ele in X[i]:
            output, hidden = rnn(Variable(ele.t()), hidden)
        loss = criterion(output, Y[i])
        
        loss.backward(retain_graph=True)
        optimizer.zero_grad()
        optimizer.step()
        print('Current loss is: ', loss)
```

The initial output was very large, then it became Inf and finally Nan. Why is this happening?

**Comment by ayush1999** [comment]
I read on the forum that it maybe because of the loss function I am using. But I get the same result even after changing it to MSELoss() or NLLLoss().

**Comment by apaszke** [comment]
It's likely an error in your data pipeline or an initialization problem. If you suspect it's a bug in a framework then please isolate the part that concerns you and let us know. If it's just a general difficulty in training your model, then you should use the forums.
